### PR TITLE
Docs: Improve Docs on mobile devices

### LINKED ISSUE #9831: Docs: Pages with horizontal scroll on mobile devices
### Description

Below documentation pages have horizontal scroll on smaller screens like mobile devices, with resolution <=414.
 [Most popular mobile](https://gs.statcounter.com/screen-resolution-stats/mobile/worldwide) resolutions are less than 414 viewport width. 
Tested on Chrome 104 / Samsung Galaxy A5 / Android 12.

A possible solution may be adding into the style

```
.table
  width: 100%
  display: block
  overflow-x: auto
```
and
```
td
  white-space: nowrap
```

### Links

**Due to the demos sticking out:**
https://handsontable.com/docs/column-hiding/ 
https://handsontable.com/docs/autocomplete-cell-type/ - demo

**Due to the table or list in the content:**

https://handsontable.com/docs/installation/
https://handsontable.com/docs/events-and-hooks/

https://handsontable.com/docs/selection/
https://handsontable.com/docs/merge-cells/
https://handsontable.com/docs/disabled-cells/

https://handsontable.com/docs/cell-renderer/
https://handsontable.com/docs/cell-editor/
https://handsontable.com/docs/cell-validator/
https://handsontable.com/docs/cell-type/
https://handsontable.com/docs/checkbox-cell-type/

https://handsontable.com/docs/context-menu/
https://handsontable.com/docs/undo-redo/
https://handsontable.com/docs/keyboard-shortcuts/
https://handsontable.com/docs/searching-values/

https://handsontable.com/docs/layout-direction/
https://handsontable.com/docs/modules/
https://handsontable.com/docs/custom-builds/
https://handsontable.com/docs/testing/
https://handsontable.com/docs/bundle-size/
https://handsontable.com/docs/third-party-licenses/
https://handsontable.com/docs/documentation-license/
https://handsontable.com/docs/release-notes/

https://handsontable.com/docs/api/core/
https://handsontable.com/docs/api/hooks/
https://handsontable.com/docs/api/options/

https://handsontable.com/docs/api/auto-column-size/
https://handsontable.com/docs/api/auto-row-size/
https://handsontable.com/docs/api/collapsible-columns/
https://handsontable.com/docs/api/column-summary/
https://handsontable.com/docs/api/comments/
https://handsontable.com/docs/api/copy-paste/
https://handsontable.com/docs/api/copy-paste/
https://handsontable.com/docs/api/drag-to-scroll/
https://handsontable.com/docs/api/dropdown-menu/
https://handsontable.com/docs/api/export-file/
https://handsontable.com/docs/api/filters/
https://handsontable.com/docs/api/hidden-columns/
https://handsontable.com/docs/api/hidden-rows/

https://handsontable.com/docs/api/base-editor/
https://handsontable.com/docs/api/data-map/
https://handsontable.com/docs/api/ghost-table/
https://handsontable.com/docs/api/index-mapper/
https://handsontable.com/docs/api/linked-physical-index-to-value-map/
https://handsontable.com/docs/api/menu/
https://handsontable.com/docs/api/menu/
https://handsontable.com/docs/api/shortcut-context/
https://handsontable.com/docs/api/shortcut-manager/

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR improves the UI styling, content styling, and demos for mobile devices (narrow viewports). The PRs changes:

#### 1. Docs versions dropdown component 
Fix the Docs versions dropdown component by moving the component to the sidebar for narrow viewports (width < 340). 

##### Before
Kapture 2022-09-20 at 09 29 59

##### After
Kapture 2022-09-20 at 09 30 27

#### 2. Search component
Fix the Search component that overlaps the Docs versions component. The PR sets the background color to fix components blending. 

##### Before
image

##### After
image

#### 3. Improve styling for tables within the Docs content
The styles are changed as the [issue](https://github.com/handsontable/handsontable/issues/9831) suggested ([_diff_](https://github.com/handsontable/handsontable/pull/9926/files#diff-64d3e686e1f494fff70ff8aadbaafe1f7c7e7fdcbbbdf476a28321c6d0cb4f7dR231)). 

##### Before
Kapture 2022-09-20 at 09 41 39

##### After
Kapture 2022-09-20 at 09 40 43

#### 4. Fix jumping-out demos
Add missing `height: 'auto'` options to all demos except that which uses the dropdown, autocomplete, or handsontable type editors. That demos can not be changed due to [the bug](https://github.com/handsontable/handsontable/issues/8688).

_[skip changelog]_

### How has this been tested?

I tested the changes using the Chrome Toggle device toolbar.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

### Related issue(s):
1. fixes #9831

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: docs/.vuepress/theme/styles/index.styl [hunk 1]
```diff
@@ -216,15 +216,25 @@ a {
...
       border-top: none;
   }
 
+  td {
+      white-space: normal;
+
+      @media (max-width: $MQNarrow) {
+        white-space: nowrap;
+      }
+  }
```
Comment (wszymanski) [inline docs/.vuepress/theme/styles/index.styl:236]: Just a minor styling change.

```suggestion
  td {
    white-space: normal;

    @media (max-width: $MQNarrow) {
      white-space: nowrap;
    }
  }
```

**Review by wszymanski — CHANGES_REQUESTED** [review]
Looks good, just one problem found. I can't see the searchbar. Please verify the issue.

Screenshot 2022-09-20 at 13 44 01

Screenshot 2022-09-20 at 13 44 01

Screenshot 2022-09-20 at 13 44 01

Device: OnePlus Nord 2 5G, Android 11

**Comment by budnix** [comment]
> Looks good, just one problem found. I can't see the searchbar. Please verify the issue.

The issue only happens when you run Docs locally on a different port than `8080`. The issue is not related to this task.

File: docs/.vuepress/theme/styles/index.styl (hunk 1 above)
Comment (budnix) [reply docs/.vuepress/theme/styles/index.styl]: Fixed within https://github.com/handsontable/handsontable/pull/9926/commits/618545d12ff033e16dbcd0fad961da0a90b1ca07

**Review by wszymanski — CHANGES_REQUESTED** [review]
As wrote here (`I've found yet another problem related to searchbar when dark mode is enabled. A text is overlapping the bar`): https://github.com/handsontable/handsontable/pull/9926#issuecomment-1252388859

**Comment by budnix** [comment]
> As wrote here (`I've found yet another problem related to searchbar when dark mode is enabled. A text is overlapping the bar`): [#9926 (comment)](https://github.com/handsontable/handsontable/pull/9926#issuecomment-1252388859)

Thanks for spotting that. I've fixed the issue. Can you check now?

**Review by wszymanski — APPROVED** [review]

**Pull request merged**